src/main/scala/io/iohk/ethereum/db/cache/SimpleLRU.scala

```diff
@@ -0,0 +1,36 @@
+package io.iohk.ethereum.db.cache
+
+import java.util
+
+/**
+  * Simplest-possible implementation of insert-time based LRU
+  *
+  * @param maxElements - capacity of the inner map, between 0 and this value.
+  * @param ttlMillis   - entry invalidation interval
+  * @tparam K - stands for the type of the keys
+  */
+class SimpleLRU[K](maxElements: Int, ttlMillis: Long) {
```

Comment:
Could we use Guava for this? I think we have that dependency anyway. I'm not sure we should have this kind of low-level implementations in this code base.

Reply:
Yes, I would certainly use guava when the cache when we have to store some values, not just the keys.

BUT even then I'd go caffeine based on the performance results: 
https://github.com/google/guava/issues/2063

Reply:
Why not caffeine with dummy values?

Reply:
If we use the guava library we can make use of FakeTicker to advance the time at will. So in this way, we don't need to use  Thread.sleep in unit test

Reply:
@enriquerodbe @biandratti finished transition to guava :)
Upgraded that test as well.